File: Day 04 - 03.04.2019/Python-Factory(Design-Patterns)/UserFactory.py
from Admin import *
from Cashier import *
from Client import *

class UserFactory:

	@staticmethod
	def login(name, password):
		if(Admin.isValid(name, password)):
			return Admin()
		if(Cashier.isValid(name, password)):
			return Cashier()
		if(Client.isValid(name, password)):
			return Client()
		
		return None

File: Day 04 - 03.04.2019/TypeScript-Factory(Design-Patterns)/UserFactory.ts
import {Cashier} from "./Cashier";
import {Admin} from "./Admin";
import {Client} from "./Client";
import { IUser } from "./IUser";

export class UserFactory
{
	public static login(name:string,password:string):IUser
	{
		if (Admin.isValid(name, password))
			return new Admin();
		if (Cashier.isValid(name, password))
			return new Cashier();
		if (Client.isValid(name, password))
			return new Client();

		return null;
	}
}

